editor/src/components/canvas/canvas-strategies/strategies/set-padding-strategy.tsx

```diff
@@ -235,16 +293,31 @@ function paddingValueIndicatorProps(
     canvasState.startingMetadata,
     filteredSelectedElements[0],
   )
-  const currentPadding = paddingMeasurementForEdge(edgePiece, padding)
+  const currentPadding = padding[paddingPropForEdge(edgePiece)]
+
+  const delta = deltaFromEdge(drag, edgePiece)
 
   const updatedPaddingMeasurement = offsetMeasurementByDelta(
     currentPadding,
-    deltaFromEdge(drag, edgePiece) + PaddingIndictorOffset(canvasState.scale),
+    delta,
     precisionFromModifiers(interactionSession.interactionData.modifiers),
   )
 
+  const maxedPaddingMeasurement = offsetMeasurementByDelta(
+    currentPadding,
+    Math.max(-currentPadding.renderedValuePx, delta),
+    precisionFromModifiers(interactionSession.interactionData.modifiers),
+  )
+
+  const message = (): string | number => {
+    if (updatedPaddingMeasurement.renderedValuePx >= PaddingTearThreshold) {
+      return printCSSNumber(maxedPaddingMeasurement.value, null)
+    }
+    return '\u2014' // emdash
+  }
```

Comment:
It looks like this function should be shared between the padding and gap controls to ensure we keep them in sync

Reply:
and more importantly, border radius controls will be able to reuse it as well

added it!

Reply:
done!